Sources/PrimerSDK/Classes/Data Models/CardNetwork.swift

```diff
@@ -7,52 +7,265 @@
+              [622010, 622999],
+              [622018],
+              [622019, 622999],
+              [62207, 62209],
+              [622126, 622925],
+              [623, 626],
+              [6270],
+              [6272],
+              [6276],
+              [627700, 627779],
+              [627781, 627799],
+              [6282, 6289],
+              [6291],
+              [6292],
+              [810],
+              [8110, 8131],
+              [8132, 8151],
+              [8152, 8163],
+              [8164, 8171],
+            ],
+                gaps: [4, 8, 12],
+                lengths: [14, 15, 16, 17, 18, 19],
+                code: CardNetworkCode(
+                    name: "CVN",
+                    length: 3))
+        case .unknown:
+            return nil
         }
     }
     
     public var icon: UIImage? {
         return nil
     }
     
-    public init(account: String) {
-        let str = account.withoutWhiteSpace
-        if str.withoutWhiteSpace.range(of: CardNetwork.masterCard.fastIdRegex, options: .regularExpression, range: nil, locale: nil) != nil {
-            self = .masterCard
-        } else if str.withoutWhiteSpace.range(of: CardNetwork.visa.fastIdRegex, options: .regularExpression, range: nil, locale: nil) != nil {
-            self = .visa
-        } else if str.withoutWhiteSpace.range(of: CardNetwork.maestro.fastIdRegex, options: .regularExpression, range: nil, locale: nil) != nil {
-            self = .maestro
-        } else if str.withoutWhiteSpace.range(of: CardNetwork.amex.fastIdRegex, options: .regularExpression, range: nil, locale: nil) != nil {
-            self = .amex
-        } else if str.withoutWhiteSpace.range(of: CardNetwork.diners.fastIdRegex, options: .regularExpression, range: nil, locale: nil) != nil {
-            self = .diners
-        } else if str.withoutWhiteSpace.range(of: CardNetwork.discover.fastIdRegex, options: .regularExpression, range: nil, locale: nil) != nil {
-            self = .discover
-        } else if str.withoutWhiteSpace.range(of: CardNetwork.jcb.fastIdRegex, options: .regularExpression, range: nil, locale: nil) != nil {
-            self = .jcb
-        } else {
-            self = .unknown
+    static func cardNumber(_ cardnumber: String, matchesPatterns patterns: [[Int]]) -> Bool {
+        for pattern in patterns {
+            if pattern.count == 1 {
+                let patternStr = String(pattern.first!)
+                if cardnumber.withoutNonNumericCharacters.hasPrefix(patternStr) {
+                    return true
+                }
+                
+            } else if pattern.count == 2 {
+                let min = pattern.first!
+                let max = pattern[1]
+                
+                for num in min...max {
+                    let numStr = String(num)
+                    if cardnumber.withoutNonNumericCharacters.hasPrefix(numStr) {
+                        return true
+                    }
+                }
```

Comment:
[Suggestion]
Feels to me that you can simplify all of these by: 
```
if pattern.count == 1 || pattern.count == 2 {
  let min = pattern.first!
  let max = pattern.count == 2 ? pattern[1] : min

   for num in min...max {
                    let numStr = String(num)
                    if cardnumber.withoutNonNumericCharacters.hasPrefix(numStr) {
                        return true
                    }
     }
}
```

Reply:
However, there's something dangerous with these: the 0 padding.
If the pattern if `[002, 010]`, this will essentially check `2` to `10`, which is not the same!
I've not seen any card starting with `0` though so we should be fine :)

Reply:
Suggestion applied. I thought about the `0`, but then I though there're no cards starting with `0`. I could replace the patterns with strings, but I think it's not needed at the moment.